### PR TITLE
Use local HEAD commit to create tsp-location.yaml

### PR DESCRIPTION
**Pull request by raych1**
In local spec repo scenario, script would use the local cloned spec repo HEAD commit to create `tsp-location.yaml` if the file doesn't exist. Once the file is created, script would proceed the sdk generation based upon the local spec repo folder.

@weshaggard

### CODE REVIEW

File: eng/common/scripts/TypeSpec-Project-Process.ps1 [hunk 1]
```diff
@@ -165,15 +165,41 @@ if (Test-Path $tmpTspConfigPath) {
...
-    Write-Error "Failed to find tsp-location.yaml in '$sdkProjectFolder', please make sure to provide CommitHash and RepoUrl parameters along with the local path of tspconfig.yaml in order to create tsp-location.yaml."
-    exit 1
+    # try to create tsp-location.yaml using HEAD commit of the local spec repo
+    Write-Warning "Failed to find tsp-location.yaml in '$sdkProjectFolder'. Trying to create tsp-location.yaml using HEAD commit of the local spec repo then proceed the sdk generation based upon local typespecs at $specRepoRoot. Alternatively, please make sure to provide CommitHash and RepoUrl parameters when running this script."
+    # set default repo to Azure/azure-rest-api-specs
+    $repo = "Azure/azure-rest-api-specs"
+    try {
+      Push-Location $specRepoRoot
+      $CommitHash = $(git rev-parse HEAD)
+      $gitRemotes = (git remote -v)
```
Comment (weshaggard) [inline eng/common/scripts/TypeSpec-Project-Process.ps1:179]:
```suggestion
      $gitOriginUrl = (git remote get-url origin)
```

File: eng/common/scripts/TypeSpec-Project-Process.ps1 [hunk 2]
```diff
@@ -165,15 +165,41 @@ if (Test-Path $tmpTspConfigPath) {
...
+      foreach ($remote in $gitRemotes) {
+        Write-Host "Checking remote $remote"
+        if ($remote.StartsWith("origin")) {
+          if ($remote -match '(.*)?github.com:(?<repo>[^/]*/azure-rest-api-specs).git') {
```
Comment (weshaggard) [inline eng/common/scripts/TypeSpec-Project-Process.ps1:183]: Should this also support the private repo?

Comment (weshaggard) [reply eng/common/scripts/TypeSpec-Project-Process.ps1:183]: I also think the ".git" at the end is optional.

File: eng/common/scripts/TypeSpec-Project-Process.ps1 [hunk 3]
```diff
@@ -165,15 +165,41 @@ if (Test-Path $tmpTspConfigPath) {
 
 $sdkProjectFolder = ""
 if ($generateFromLocalTypeSpec) {
+  Write-Host "Generating sdk code based on local type specs at $specRepoRoot."
```
Comment (konrad-jamrozik) [inline eng/common/scripts/TypeSpec-Project-Process.ps1:168]:
```suggestion
  Write-Host "Generating sdk code based on local type specs at specRepoRoot: $specRepoRoot."
```

File: eng/common/scripts/TypeSpec-Project-Process.ps1 (hunk 2 above)
Comment (raych1) [reply eng/common/scripts/TypeSpec-Project-Process.ps1]: updated

File: eng/common/scripts/TypeSpec-Project-Process.ps1 [hunk 4]
```diff
@@ -165,15 +165,41 @@ if (Test-Path $tmpTspConfigPath) {
...
-    exit 1
+    # try to create tsp-location.yaml using HEAD commit of the local spec repo
+    Write-Warning "Failed to find tsp-location.yaml in '$sdkProjectFolder'. Trying to create tsp-location.yaml using HEAD commit of the local spec repo then proceed the sdk generation based upon local typespecs at $specRepoRoot. Alternatively, please make sure to provide CommitHash and RepoUrl parameters when running this script."
+    # set default repo to Azure/azure-rest-api-specs
+    $repo = "Azure/azure-rest-api-specs"
+    try {
+      Push-Location $specRepoRoot
+      $CommitHash = $(git rev-parse HEAD)
+      $gitOriginUrl = (git remote get-url origin)
+      foreach ($remote in $gitOriginUrl) {
```
Comment (weshaggard) [inline eng/common/scripts/TypeSpec-Project-Process.ps1:180]: You don't need this loop now only the loop any more only the regex match

**Comment by raych1** [comment]
@weshaggard , can you approve this PR?

**Review by weshaggard — APPROVED** [review]

**Pull request merged**